database/src/adapters/receipts.rs

```diff
@@ -486,14 +486,43 @@ async fn save_receipts(
 
 async fn store_receipt_actions(
     pool: &actix_diesel::Database<PgConnection>,
-    receipts: Vec<&near_indexer_primitives::views::ReceiptView>,
+    receipts: &[&near_indexer_primitives::views::ReceiptView],
     block_timestamp: u64,
+) -> anyhow::Result<()> {
+    try_join!(
+        store_action_receipts(pool, receipts),
+        store_action_receipt_actions(pool, receipts, block_timestamp),
+        store_action_receipt_input_data(pool, receipts),
+        store_action_receipt_output_data(pool, receipts),
+    )?;
+    Ok(())
+}
+
+async fn store_action_receipts(
+    pool: &actix_diesel::Database<PgConnection>,
+    receipts: &[&near_indexer_primitives::views::ReceiptView],
 ) -> anyhow::Result<()> {
     let receipt_actions: Vec<models::ActionReceipt> = receipts
         .iter()
         .filter_map(|receipt| models::ActionReceipt::try_from(*receipt).ok())
         .collect();
+    crate::await_retry_or_panic!(
+        diesel::insert_into(schema::action_receipts::table)
+            .values(receipt_actions.clone())
+            .on_conflict_do_nothing()
+            .execute_async(pool),
+        10,
+        "ReceiptActions were stored in database".to_string(),
```

Comment:
https://github.com/near/near-indexer-for-explorer/blob/30f3d407c1ac8a76e3659466b974dc4060965521/database/src/retryable.rs#L2-L3

This is actually an error message, maybe this wording makes more sense? Same for all other cases below

```suggestion
        "Failed to store ReceiptActions in database".to_string(),
```

Reply:
Agree, I saw many places where the naming/phrases could be improved, just decided not to dig too deep into this :)

Reply:
😅 offtopic: I wish someone could explain to me why did I do such bad naming in the first place. There definitely was a reason I don't recall.